WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetailsViewModel.swift

```diff
@@ -93,11 +98,46 @@ final class ShippingLabelPackageDetailsViewModel: ObservableObject {
         self.weightUnit = weightUnit
         self.packagesResponse = packagesResponse
         self.selectedPackageID = selectedPackageID
+        self.initialTotalWeight = totalWeight
+
+        configureItemRows()
+        configureTotalWeights()
```

Comment:
Isn't possible here to pass the `totalWeight` directly to the method, without declaring a new variable `initialTotalWeight`?

Reply:
This variable `initialTotalWeight` honestly seems like a hack since we only use it once to check if the weight was input manually initially, and then it is set to nil again (please look at line 129). 

I can't use `totalWeight` because my intention is to only check the initial weight if any; and if the initial weight is different from calculated weight, we can safely say that the weight was input manually and switching packages should not affect the weight.

Since you're confused with this, I'm afraid it's not the best way to handle it. I'm not sure of a better way though - any suggestion?

Reply:
Instead of using `initialTotalWeight`, we can check if `isPackageWeightEdited` is `false`, or I'm wrong? In this case, we are sure that the initial total weight never changed, on the contrary, if there is a change, we will save the new total weight and we will set `isPackageWeightEdited` to `true` (like the current logic).

Reply:
The reason why I think `initialTotalWeight` is necessary is in [this discussion](https://github.com/woocommerce/woocommerce-ios/pull/4658#discussion_r673653542). Basically we have to compare the input weight with the calculated one to see if user has manually edited the weight - does that make sense?